File: Masters degree project application/Sklep/src/main/java/pl/kul/Sklep/Entity/Order.java
package pl.kul.Sklep.Entity;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Document(collection = "orders")
public class Order {
    @Id
    private String id;
    private String userName;
    private String userLastName;
    private String userEmail;
    private String userPhone;
    private List<OrderItem> items;
    private Address shippingAddress;
    private double totalPrice;
    private LocalDateTime orderDate;
    private String status;
    private PaymentInfo paymentInfo;
}

File: Masters degree project application/frontend/src/app/models/order.ts
export class Order {
  constructor(
    public id: string | null = null,
    public userName: string,
    public userLastName: string,
    public userEmail: string,
    public userPhone: string,
    public items: {
      productId: string;
      quantity: number;
      price: number;
    }[] = [],
    public totalPrice: number = 0,
    public shippingAddress: {
      street: string;
      city: string;
      postalCode: string;
      country: string;
    },
    public orderDate: Date | null = null,
    public status: string = 'Pending'
  ) {}
}
